classifier/firmware_test.go

```diff
@@ -0,0 +1,162 @@
+package classifier
+
+import (
+	"testing"
+
+	bmh "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/stretchr/testify/assert"
+
+	hwcc "github.com/metal3-io/hardware-classification-controller/api/v1alpha1"
+)
+
+func TestFirmwareDetails(t *testing.T) {
+	assert.True(t, checkString("Dell Inc.", "Dell Inc."))
+	assert.True(t, checkString("1.5.6", "1.5.6"))
+	assert.True(t, checkString("", "Dell Inc."))
+	assert.True(t, checkString("", "1.5.6"))
+	assert.False(t, checkString("1.5", "1.5.6"))
+	assert.False(t, checkString("Dell", "Dell Inc."))
+
+}
+
+func TestFirmwareVendor(t *testing.T) {
+	testCases := []struct {
+		Scenario string
+		Rule     *hwcc.Firmware
+		Actual   string
+		Expected bool
+	}{
+		{
+			Scenario: "nil",
+			Rule:     nil,
+			Actual:   "Dell Inc.",
+			Expected: true,
+		},
+		{
+			Scenario: "matched",
+			Rule: &hwcc.Firmware{
+				Vendor: "Dell Inc.",
+			},
+			Actual:   "Dell Inc.",
+			Expected: true,
+		},
+		{
+			Scenario: "unmatched",
+			Rule: &hwcc.Firmware{
+				Vendor: "Dell Inc.",
+			},
+			Actual:   "Dell",
+			Expected: false,
+		},
+		{
```

Comment:
If you write a test that sets vendor to "" and version to a valid match, you'll see what I mean by my earlier comment.

Reply:
Actually its now independent of each other, So if user doesn't specify the version we can still run check on vendor,
I have added the unit test cases for the same.